### PR TITLE
improve boosting overfit

### PR DESCRIPTION
**Pull request by noamzbr**

### CODE REVIEW

File: deepchecks/checks/methodology/boosting_overfit.py [hunk 1]
```diff
@@ -118,7 +118,7 @@ def calculate_steps(num_steps, num_estimators):
 
 
 class BoostingOverfit(TrainTestBaseCheck):
-    """Check for overfit occurring when increasing the number of iterations in boosting models.
+    """Check for overfit caused by using too many boosting iterations in a gradient boosted model.
```
Comment (nirhutnik) [inline deepchecks/checks/methodology/boosting_overfit.py:121]: "boosting iterations" is a bit weird, especially when you later affirm that you're talking about gradient boosting.
WDYT?

File: deepchecks/checks/methodology/boosting_overfit.py [hunk 2]
```diff
@@ -171,6 +171,7 @@ def _boosting_overfit(self, train_dataset: Dataset, test_dataset: Dataset, model
         else:
             metric_name = DEFAULT_SINGLE_METRIC[model_type]
             scorer = DEFAULT_METRICS_DICT[model_type][metric_name]
+            metric_name = metric_name + ' (Default)'
```
Comment (nirhutnik) [inline deepchecks/checks/methodology/boosting_overfit.py:174]: Why not just add this to line 172?

File: deepchecks/checks/methodology/boosting_overfit.py [hunk 3]
```diff
@@ -193,8 +194,13 @@ def display_func():
             # Display x ticks as integers
             axes.xaxis.set_major_locator(MaxNLocator(integer=True))
 
+        display_text = f"""<span>
+            The check is done by limiting the boosting model to use up to X estimators ech time, and plotting the
```
Comment (nirhutnik) [inline deepchecks/checks/methodology/boosting_overfit.py:198]:
```suggestion
            The check limits the boosting model to using up to N estimators each time, and plotting the
```

**Review by nirhutnik — CHANGES_REQUESTED** [review]

File: deepchecks/checks/methodology/boosting_overfit.py (hunk 2 above)
Comment (noamzbr) [reply deepchecks/checks/methodology/boosting_overfit.py:174]: because then line 173 fails

File: deepchecks/checks/methodology/boosting_overfit.py (hunk 1 above)
Comment (noamzbr) [reply deepchecks/checks/methodology/boosting_overfit.py:121]: mmm not sure I understand why that is weird. What is the alternative you suggest?

File: deepchecks/checks/methodology/boosting_overfit.py (hunk 2 above)
Comment (nirhutnik) [reply deepchecks/checks/methodology/boosting_overfit.py:174]: Oh. Ok

File: deepchecks/checks/methodology/boosting_overfit.py (hunk 1 above)
Comment (nirhutnik) [reply deepchecks/checks/methodology/boosting_overfit.py:121]: Dropping the first use of "boosting". Just saying "too many iterations"

**Review by nirhutnik — APPROVED** [review]
